src/test/architecture/test_architecture.py

```diff
@@ -312,3 +312,146 @@ def test_verify_callback_uniqueness(self, mocker, caplog):
         caplog.clear()
         Architecture._verify([node_mock])
         assert len(caplog.record_tuples) == 1
+
+    def test_rename_function(self, mocker):
+        # define test case: path
+        architecture_text = """
+named_paths:
+- path_name: target_path_0
+  node_chain:
+  - node_name: /node_0
+    publish_topic_name: /topic_0
+  - node_name: /node_1
+    subscribe_topic_name: /topic_0
+- path_name: target_path_1
+  node_chain:
+  - node_name: /node_1
+    publish_topic_name: /topic_1
+  - node_name: /node_0
+    subscribe_topic_name: /topic_1"""
+
+        # define test case: exec
+        architecture_text += """
+executors:
+- executor_type: single_threaded_executor
+  executor_name: executor_0
+  callback_group_names:
+  - /callback_group_0
+  - /callback_group_1
+- executor_type: single_threaded_executor
+  executor_name: executor_1
+  callback_group_names:
+  - /callback_group_0
+  - /callback_group_1"""
+
+        # define test case: node, callback, topic
+        architecture_text += """
+nodes:
+- node_name: /node_0
+  callback_groups:
+  - callback_group_type: mutually_exclusive
+    callback_group_name: /callback_group_1
+    callback_names:
+    - /callback_0
+    - /callback_1
+  callbacks:
+  - callback_name: /callback_0
+    callback_type: timer_callback
+    period_ns: 200000000
+    symbol: timer
+  - callback_name: /callback_1
+    callback_type: subscription_callback
+    topic_name: /topic_1
+    symbol: sub
+  publishes:
+  - topic_name: /topic_0
+    callback_names:
+    - /callback_0
+  subscribes:
+  - topic_name: /topic_1
+    callback_name: /callback_1"""
+
+        # define test case: node, callback, topic
+        architecture_text += """
+- node_name: /node_1
+  callback_groups:
+  - callback_group_type: mutually_exclusive
+    callback_group_name: /callback_group_0
+    callback_names:
+    - /callback_2
+    - /callback_3
+  callbacks:
+  - callback_name: /callback_2
+    callback_type: timer_callback
+    period_ns: 200000000
+    symbol: timer
+  - callback_name: /callback_3
+    callback_type: subscription_callback
+    topic_name: /topic_0
+    symbol: sub
+  publishes:
+  - topic_name: /topic_1
+    callback_names:
+    - /callback_2
+  subscribes:
+  - topic_name: /topic_0
+    callback_name: /callback_3
+        """
```

Comment:
Test case have:
- node_0, node_1
- executor_0, executor_1
- topic_0(from node_0 to node_1), topic_1(from node_1 to node_0)
- named_paths_0(from node_0 to node_1 with topic0), named_path_1(from node_1 to node_0 with topic_1)
- callback_0(publish topic_0), callback_1(subscribe topic_1), callback_2(publish topic_1), callback_3(subscribe topic_0)